testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java

```diff
@@ -366,7 +383,18 @@ public static void inIndependentFactories(int numThreads, int timeoutSeconds, Ru
             );
         } finally {
             LOG.debugf("waiting for threads to shutdown to avoid that one test pollutes another test");
-            es.shutdown();
+            es.shutdownNow();
+            if(!es.awaitTermination(10, TimeUnit.SECONDS)) {
+                List<String> liveStacks = threads.stream().filter(Thread::isAlive).map(thread -> {
+                    StringBuilder sb = new StringBuilder();
+                    sb.append(thread.getName()).append(":\n");
+                    for (StackTraceElement traceElement : thread.getStackTrace()) {
+                        sb.append("\tat ").append(traceElement).append("\n");
+                    }
+                    return sb.toString();
```

Comment:
Can be replaced with a less talkative stack trace, see `StackUtil` methods

Reply:
@hmlnarik - I had a look at `StackUtil` and it only reads the current thread's stack trace. Here I'm interested in other threads stack traces. Therefore I'd like to keep the code. In the most recent version I'm adding more information about other threads running on the JVM.

Reply:
The code evolved from this even further. As the code it relates to is obsolete, I resolve this.